app/helpers/bootstrap_form_builder.rb

```diff
@@ -0,0 +1,21 @@
+class BootstrapFormBuilder < ActionView::Helpers::FormBuilder
+  %w[text_field text_area file_field].each do |method_name|
+    define_method(method_name) do |attribute, options = {}|
+      feedback_name = @object_name + '-' + attribute.to_s + '-feedback'
+      options["aria-describedby"] = feedback_name
+      options[:class] = options[:class].present? ? "#{options[:class]} form-control" : 'form-control'
+      unless @object[attribute].nil?
+        options[:class] += (@object.errors[attribute].any? ? ' is-invalid' : ' is-valid')
+      end
+
+      @template.content_tag :div, class: 'form-group' do
+        label(attribute, class: 'form-label') + super(attribute, options) +
+          @template.content_tag(:div, class: 'invalid-feedback', id: feedback_name) do
```

Comment:
Ar šo bloku viss ir viltigak. Tev nevajag viņu renderet ja nav kļudu.
Es paskatijos kā tas izskatas portalā. Bootstrap it kā slepj to, jā nav klases invalid.
Bet tas ir tikai ar bootstrap.

Pielabo, lai viņš renderetos tikai kad ir kļudas un labak iznes to atseviškā metode:
```ruby
  def errors_block(attribute, feedback_name)
    return if @object.errors[attribute].empty?

    @template.tag.div class: 'invalid-feedback', id: feedback_name do

```